app/controllers/projects/blob_controller.rb

```diff
@@ -146,10 +146,19 @@ def editor_variables
 
     @file_path =
       if action_name.to_s == 'create'
+        if params[:file_upload].present?
+          params[:file_name] = params[:file_upload].original_filename
+        end
         File.join(@path, File.basename(params[:file_name]))
       else
         @path
       end
+     
+    if params[:file_upload].present?
+      params[:content] = params[:file_upload].read
+      params[:encoding] = 'text'
```

Comment:
@stanhu sorry write back to you late. I tested the file replacement, did you compare the raw file to the original image file?
I replaced "publicmilestone.png" with "gitlab_diagram_overview.png", and compared the raw file of replaced "publicmilestone.png" with "gitlab_diagram_overview.png", they are the same.
![compare_hex](https://cloud.githubusercontent.com/assets/7370334/8849281/c9b840fa-313e-11e5-9c30-998741331d89.png)
However, in the view, the size of the file used to replace is not shown correctly, as it is considered as 'text' encoded, right?
original file in local directory
![original](https://cloud.githubusercontent.com/assets/7370334/8849407/a79060e2-313f-11e5-9698-e6310c08e6f1.png)
replace in view (after being download, this file has the same size and content as original)
![replace](https://cloud.githubusercontent.com/assets/7370334/8849406/a78f603e-313f-11e5-9046-95da24da6f4c.png)